engine/script/src/script_sys.cpp

```diff
@@ -342,87 +342,149 @@ union SaveLoadBuffer
         return 1;
     }
 
-    /*# get config value
-     * Get config value from the game.project configuration file.
-     *
-     * In addition to the project file, configuration values can also be passed
-     * to the runtime as command line arguments with the `--config` argument.
+    static dmConfigFile::HConfig GetConfigFile(lua_State* L)
+    {
+        HContext context = dmScript::GetScriptContext(L);
+        if (context)
+        {
+            return context->m_ConfigFile;
+        }
+        return 0;
+    }
+
+    /*# get string config value with optional default value
+     * Get string config value from the game.project configuration file with optional default value
      *
-     * @name sys.get_config
+     * @name sys.get_config_string
      * @param key [type:string] key to get value for. The syntax is SECTION.KEY
-     * @return value [type:string] config value as a string. nil if the config key doesn't exists
+     * @param [default_value] [type:string] (optional) default value to return if the value does not exist
+     * @return value [type:string] config value as a string. default_value if the config key does not exist. nil if no default value was supplied.
      * @examples
      *
-     * Get display width
+     * Get user config value
      *
      * ```lua
-     * local width = tonumber(sys.get_config("display.width"))
+     * local text = sys.get_config_string("my_game.text", "default text"))
      * ```
      *
-     * Start the engine with a bootstrap config override and a custom config value
+     * Start the engine with a bootstrap config override and add a custom config value
      *
      * ```
-     * $ mygame --config=bootstrap.main_collection=/mytest.collectionc --config=mygame.testmode=1
+     * $ dmengine --config=bootstrap.main_collection=/mytest.collectionc --config=mygame.testmode=1
      * ```
      *
-     * Set and read a custom config value
+     * Read the custom config value from the command line
      *
      * ```lua
-     * local testmode = tonumber(sys.get_config("mygame.testmode"))
+     * local testmode = sys.get_config_int("mygame.testmode")
      * ```
      */
+    static int Sys_GetConfigString(lua_State* L)
```

Comment:
When implementing the actual native extension, it became clear that it wasn't possible to properly override the settings, because the "GetString()" was the only function getting called.

So, I created these variants instead, and deprecated (removed documentation) for the sys.get_config()